flink-runtime/src/main/java/org/apache/flink/runtime/akka/FlinkUntypedActor.java

```diff
@@ -0,0 +1,140 @@
+ * detected, then an Exception is thrown.
+ *
+ * In order to implement the actor behavior, an implementing subclass has to override the method
+ * handleMessage, which defines how messages are processed. Furthermore, the subclass has to provide
+ * a leader session ID option which is returned by getLeaderSessionID.
+ */
+abstract public class FlinkUntypedActor extends UntypedActor {
+	protected static Logger LOG = LoggerFactory.getLogger(FlinkUntypedActor.class);
+
+	/**
+	 * This method is called by Akka if a new message has arrived for the actor. It logs the
+	 * processing time of the incoming message if the logging level is set to debug. After logging
+	 * the handleLeaderSessionID method is called.
+	 *
+	 * Important: This method cannot be overriden. The actor specific message handling logic is
+	 * implemented by the method handleMessage.
+	 *
+	 * @param message Incoming message
+	 * @throws Exception
+	 */
+	@Override
+	public final void onReceive(Object message) throws Exception {
+		if(LOG.isDebugEnabled()) {
+			LOG.debug("Received message {} at {} from {}.", message, getSelf().path(), getSender());
+
+			long start = System.nanoTime();
+
+			handleLeaderSessionID(message);
+
+			long duration = (System.nanoTime() - start)/ 1000000;
+
+			LOG.debug("Handled message {} in {} ms from {}.", message, duration, getSender());
+		} else {
+			handleLeaderSessionID(message);
+		}
+	}
+
+	/**
+	 * This method filters out {@link LeaderSessionMessage} whose leader session ID is not equal
+	 * to the actors leader session ID. If a message of type {@link RequiresLeaderSessionID}
+	 * arrives, then an Exception is thrown, because these messages have to be wrapped in a
+	 * {@link LeaderSessionMessage}.
+	 *
+	 * @param message Incoming message
+	 * @throws Exception
+	 */
+	private void handleLeaderSessionID(Object message) throws Exception {
+		if(message instanceof LeaderSessionMessage) {
+			LeaderSessionMessage msg = (LeaderSessionMessage) message;
+
+			if(msg.leaderSessionID().isDefined() && getLeaderSessionID().isDefined()) {
+				if(getLeaderSessionID().equals(msg.leaderSessionID())) {
+					// finally call method to handle message
+					handleMessage(msg.message());
+				} else {
+					handleDiscardedMessage(msg);
+				}
+			} else {
+				handleDiscardedMessage(msg);
+			}
+		} else if (message instanceof RequiresLeaderSessionID) {
+			throw new Exception("Received a message " + message + " without a leader session " +
+					"ID, even though it requires to have one.");
+		} else {
+			// call method to handle message
+			handleMessage(message);
+		}
+	}
+
+	private void handleDiscardedMessage(Object msg) {
+		LOG.debug("Discard message {} because the leader session ID was not correct.", msg);
+	}
+
+	/**
+	 * This method contains the actor logic which defines how to react to incoming messages.
+	 *
+	 * @param message Incoming message
+	 * @throws Exception
+	 */
+	abstract protected void handleMessage(Object message) throws Exception;
```

Comment:
The style in all the other code is to have first the visibility, then the abstract modifier. I think we should follow one consistent pattern in the project.